### PR TITLE
Feature/firebase

### PR DESCRIPTION
**Pull request by Thee-Dust**
## Description
Please include a summary of the change and which issue is fixed. Please also include relevant screenshots of your code being deployed. 
- adds firebase and full  functionality to login/signup and logout

Fixes # (issue)
- #

## Type of change
Please delete options that are not relevant.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Refactoring

## Checklist:
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

File: src/components/App/App.js [hunk 1]
```diff
@@ -41,26 +43,31 @@ class App extends React.Component {
...
-        render={({match}) => {
-          const id = match.params.id;
-          return <PokemonDetails id={id} caught={caughtPokemon} favorite={this.catchPokemon} />
-        }}/>
-        </Switch>
+        <AuthProvider>
+          <Switch>
+            <Route path="/signup" component={Signup} />
+            <Route path="/login" component={Login} />
+            <PrivateRoute
```
Comment (rhen92) [inline src/components/App/App.js:58]: I really like the use of this Private Route component only allowing the user to see things once they are logged in!

File: src/components/App/App.js [hunk 2]
```diff
@@ -49,37 +51,31 @@ class App extends React.Component {
...
+              component={Home}
+              pokemons={pokemons}
+              caught={caughtPokemon} 
+              favorite={this.catchPokemon} 
+            />
+            <PrivateRoute path='/caught' 
+              component={Caught} 
+              pokemons={pokemons} 
+              caught={caughtPokemon} 
+              favorite={this.catchPokemon}
```
Comment (asiisii) [inline src/components/App/App.js:69]: Great use of PrivateRoute

File: src/components/Login/Login.js [hunk 3]
```diff
@@ -1,45 +1,54 @@
...
-//     </section>
-//   )
-// }
+export default function Login() {
+  const emailRef = useRef()
+  const passwordRef = useRef()
+  const { login } = useAuth()
+  const [error, setError] = useState('')
+  const [loading, setLoading] = useState('')
+  const history = useHistory()
```
Comment (asiisii) [inline src/components/Login/Login.js:11]: nice use of useRef() hooks

File: src/components/Login/Login.js [hunk 4]
```diff
@@ -1,45 +1,54 @@
...
 
+  async function handleSubmit(e) {
+    e.preventDefault()
+    try{
+      setError('')
+      setLoading(true)
+      await login(emailRef.current.value, passwordRef.current.value)
+      history.push('/')
```
Comment (asiisii) [inline src/components/Login/Login.js:19]: oh just learned a new thing

File: src/components/PrivateRoute/PrivateRoute.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import React from 'react'
+import { Route, Redirect } from 'react-router-dom'
+import { useAuth } from '../../context/AuthContext'
+import Home from '../Home/Home'
+
+export default function PrivateRoute({component: Component, ...rest}) {
```
Comment (rhen92) [inline src/components/PrivateRoute/PrivateRoute.js:6]: Love how reusable this component is!!

File: src/components/PrivateRoute/PrivateRoute.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+  const { currentUser } = useAuth()
+  return (
+    <Route
+      render={() => {
+        const id = rest.computedMatch.params.id
+        return currentUser ? <Component id={id} { ...rest} /> : <Redirect to='/login' />
+      }}>
+    </Route>
```
Comment (asiisii) [inline src/components/PrivateRoute/PrivateRoute.js:14]: love it! super dynamic!!

File: src/components/Signup/Signup.js [hunk 7]
```diff
@@ -1,102 +1,63 @@
...
-        <form>
-          <label>Username</label>
-          <input 
-            type='email' 
-            required 
-            value={this.state.email} 
-            onChange={e => this.setEmail(e)}
-          />
+    if(passwordRef.current.value !== passwordConfirmationRef.current.value){
+      return setError('Passwords do not match')
```
Comment (rhen92) [inline src/components/Signup/Signup.js:17]: I like this check here!

**Review by asiisii — APPROVED** [review]

**Pull request merged**